clang/lib/Basic/Targets/WebAssembly.h

```diff
@@ -90,6 +90,7 @@ class LLVM_LIBRARY_VISIBILITY WebAssemblyTargetInfo : public TargetInfo {
 
   StringRef getABI() const override;
   bool setABI(const std::string &Name) override;
+  bool useFP16ConversionIntrinsics() const override { return false; }
```

Comment:
Is this used?

Reply:
Yeah, this is what causes clang to start outputting `half` types. I could conditionally enable this with `return !HasHalfPrecision;` instead. Though doing a quick test with scalar `__fp16` in c, the `half` types seems to work correctly regardless of this setting.

Reply:
I am not very familiar with the places where the function is used, but `!HasHalfPrecision` seems safer.